src/utils/setup/astroport.rs

```diff
@@ -0,0 +1,264 @@
+            .code_id
+            .ok_or(Error::MissingContextVariable(String::from(
+                "astroport_token_registry::code_id",
+            )))?;
+
+        let contract = contract_a.instantiate(
+            key,
+            serde_json::json!({
+                "owner": owner_addr,
+            })
+            .to_string()
+            .as_str(),
+            TOKEN_REGISTRY_NAME,
+            None,
+            "",
+        )?;
+        let addr = contract.address;
+        let artifact_path =
+            contract_a
+                .file_path
+                .ok_or(Error::MissingContextVariable(String::from(
+                    "astroport_token_registry::artifact_path",
+                )))?;
+
+        let neutron = self.get_mut_chain(NEUTRON_CHAIN_ID);
+
+        neutron
+            .contract_addrs
+            .entry(TOKEN_REGISTRY_NAME.to_owned())
+            .or_default()
+            .push(addr.clone());
+
+        self.astroport_token_registry = Some(DeployedContractInfo {
+            code_id,
+            address: addr,
+            artifact_path,
+        });
+
+        Ok(())
+    }
+
+    /// Instantiates the astroport factory.
+    /// Note: by default, all pair types are enabled
+    pub fn tx_create_factory(&mut self, key: &str, factory_owner: &str) -> Result<(), Error> {
+        let neutron = self.get_chain(NEUTRON_CHAIN_ID);
+
+        let pair_xyk_code_id =
+            neutron
+                .contract_codes
+                .get(PAIR_NAME)
+                .ok_or(Error::MissingContextVariable(String::from(
+                    "contract_codes::astroport_pair",
+                )))?;
+        let pair_stable_code_id =
+            neutron
+                .contract_codes
+                .get(STABLE_PAIR_NAME)
+                .ok_or(Error::MissingContextVariable(String::from(
+                    "contract_codes::astroport_pair_stable",
+                )))?;
+        let token_code_id =
+            neutron
+                .contract_codes
+                .get(TOKEN_NAME)
+                .ok_or(Error::MissingContextVariable(String::from(
+                    "contract_codes::cw20_base",
+                )))?;
+        let whitelist_code_id =
+            neutron
+                .contract_codes
+                .get(WHITELIST_NAME)
+                .ok_or(Error::MissingContextVariable(String::from(
+                    "contract_codes::astroport_whitelist",
+                )))?;
+
+        let native_registry_addr = neutron
+            .contract_addrs
+            .get(TOKEN_REGISTRY_NAME)
+            .and_then(|maybe_addr| maybe_addr.get(0))
+            .ok_or(Error::MissingContextVariable(String::from(
+                "contract_ddrs::astroport_native_coin_registry",
+            )))?;
+
+        let mut contract_a = self.get_contract(FACTORY_NAME)?;
+
+        let contract = contract_a.instantiate(
+            key,
+            serde_json::json!({
+                "pair_configs": [
```

Comment:
I suggest adding the current astroport rev as a dependency and use the message there instead of writing the json directly here. Something like this https://github.com/timewave-computer/ticker-bot/blob/9d381545db907317d929e89d91f9be5b382d2dd8/local-interchaintest/src/main.rs#L100. This way if we update the astroport version that we are using, it will automatically detect any new/old fields that we need to add/remove. It will also make sure that you are don't make a mistake creating the msg. Like ideally never write jsons but fill in the Msgs and use serde_json to convert them. WDYT?

Reply:
Good idea!

Reply:
I implemented this! Now all Astroport calls use the underlying messages!